lib/rules/no-misleading-character-class.js

```diff
@@ -193,6 +200,28 @@ const findCharacterSequences = {
 
 const kinds = Object.keys(findCharacterSequences);
 
+/**
+ * Gets the value of the given node if it's a static value other than a regular expression object,
+ * or the node's `regex` property.
+ * The purpose of this method is to provide a replacement for `getStaticValue` on Node.js 18, where `getStaticValue` returns `null` if a regular expression literal contains the `v` flag.
+ * A limitation of this method is that it can only detect a regular expression if the specified node is itself a regular expression literal node.
+ * @param {ASTNode} node The node to be inspected.
+ * @param {Scope} [initialScope] Optional scope to start finding variables. If this scope was given, this tries to resolve identifier references which are in the given node as much as possible.
+ * @returns {{ value: any } | { value: undefined, optional?: true } | { regex: { pattern: string, flags: string } } | null} The static value of the node, or `null`.
```

Comment:
In what cases does this function return `optional: true` property?

Reply:
Interesting, it looks like this property was added for internal calculations rather than as something that consumers might make use of, and it isn't documented. Either way, if it can't happen in our case, it might be better to remove that typing.

Reply:
Yes, I only noticed that because of [the JSDoc](
https://github.com/eslint-community/eslint-utils/blob/0540eb3435a80cce4a02c663833d800812dcff51/src/get-static-value.mjs#L630C43-L630C57) in the `@return` tag. Updated in 64813fbc71b5669e5d0cbe33bf64fcd78c069b47.